internal/advertisement-operator/controller.go

```diff
@@ -302,8 +326,129 @@ func (r *AdvertisementReconciler) cleanOldAdvertisements() {
 				if err := r.Client.Delete(context.Background(), &adv, &client.DeleteOptions{}); err != nil {
 					klog.Error(err)
 				}
+				klog.Infof("Adv %v expired. TimeToLive was %v", adv.Name, adv.Spec.TimeToLive)
 			}
 		}
 		time.Sleep(10 * time.Minute)
 	}
 }
+
+func (r *AdvertisementReconciler) checkClusterStatus(adv protocolv1.Advertisement) error {
+	// get the kubeconfig provided by the foreign cluster
+	remoteKubeconfig, err := r.AdvClient.Client().CoreV1().Secrets(adv.Spec.KubeConfigRef.Namespace).Get(context.Background(), adv.Spec.KubeConfigRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// METHOD 1 : kubectl get pods (on foreign cluster)
+	remoteClient, err := protocolv1.CreateAdvertisementClient("", remoteKubeconfig)
+	if err != nil {
+		return err
+	}
+	retry := 0
+	for {
+		_, err = remoteClient.Client().CoreV1().Pods("default").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			retry++
+		} else {
+			retry = 0
+		}
+		if retry == 3 {
+			return err
+		}
+		time.Sleep(20 * time.Second)
+	}
+
+	// METHOD 2 : CRONJOB
```

Comment:
Deadcode should be put in a dedicated commit if we want to keep it and then removed.